src/actions/charts.ts

```diff
@@ -0,0 +1,63 @@
+import { Dispatch } from 'redux';
+import { createAction, getReturnOfExpression } from 'typesafe-actions';
+
+import { StoreState, Chart } from '../shared/types';
+import * as url from '../shared/url';
+
+export const requestCharts = createAction('REQUEST_CHARTS');
+export const receiveCharts = createAction('RECEIVE_CHARTS', (charts: Chart[]) => ({
+  type: 'RECEIVE_CHARTS',
+  charts
+}));
+export const selectChart = createAction('SELECT_CHART', (chart: Chart) => ({
+  type: 'SELECT_CHART',
+  chart
+}));
+
+const allActions = [requestCharts, receiveCharts, selectChart].map(getReturnOfExpression);
+export type ChartsAction = typeof allActions[number];
+
+export function fetchCharts(repo: string) {
+  return (dispatch: Dispatch<StoreState>): Promise<{}> => {
```

Comment:
Might want to see if we can add some return typings.

Reply:
Do you mean the type of the promise? I left it as `{}` for now since it's not being chained by anything. If I understand it correctly, this would cause a type error when we try to access the object at which point we will need to set the type correctly?